package.json

```diff
@@ -39,17 +39,21 @@
     "tslib": "^2.4.1",
     "typescript": "^4.9.4",
     "vite": "^4.0.0",
+    "vite-plugin-top-level-await": "^1.2.2",
```

Comment:
We could get rid of this by:

```diff
diff --git a/package-lock.json b/package-lock.json
index 40c77c3..2c61e86 100644
--- a/package-lock.json
+++ b/package-lock.json
@@ -53,7 +53,6 @@
         "tslib": "^2.4.1",
         "typescript": "^4.9.4",
         "vite": "^4.0.0",
-        "vite-plugin-top-level-await": "^1.2.2",
         "vitest": "^0.25.7"
       },
       "engines": {
@@ -1307,195 +1306,6 @@
         "vite": "^4.0.0"
       }
     },
-    "node_modules/@swc/core": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core/-/core-1.3.22.tgz",
-      "integrity": "sha512-oQ9EPEb7NgWcGIDoVfLCuffvtC4MzVtrwjqwKzFHP8FUh1fn8+2wraOjkkDXW74BB4Hgve5ykkaHix9bebB9Ww==",
-      "dev": true,
-      "hasInstallScript": true,
-      "bin": {
-        "swcx": "run_swcx.js"
-      },
-      "engines": {
-        "node": ">=10"
-      },
-      "funding": {
-        "type": "opencollective",
-        "url": "https://opencollective.com/swc"
-      },
-      "optionalDependencies": {
-        "@swc/core-darwin-arm64": "1.3.22",
-        "@swc/core-darwin-x64": "1.3.22",
-        "@swc/core-linux-arm-gnueabihf": "1.3.22",
-        "@swc/core-linux-arm64-gnu": "1.3.22",
-        "@swc/core-linux-arm64-musl": "1.3.22",
-        "@swc/core-linux-x64-gnu": "1.3.22",
-        "@swc/core-linux-x64-musl": "1.3.22",
-        "@swc/core-win32-arm64-msvc": "1.3.22",
-        "@swc/core-win32-ia32-msvc": "1.3.22",
-        "@swc/core-win32-x64-msvc": "1.3.22"
-      }
-    },
-    "node_modules/@swc/core-darwin-arm64": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-darwin-arm64/-/core-darwin-arm64-1.3.22.tgz",
-      "integrity": "sha512-MMhtPsuXp8gpUgr9bs+RZQ2IyFGiUNDG93usCDAFgAF+6VVp+YaAVjET/3/Bx5Lk2WAt0RxT62C9KTEw1YMo3w==",
-      "cpu": [
-        "arm64"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "darwin"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-darwin-x64": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-darwin-x64/-/core-darwin-x64-1.3.22.tgz",
-      "integrity": "sha512-SG6QbNat4GZ5VJU3Zo6a54oQOtbhJVE6BCQw4JjOCZJmAeBzNebGy9wsT4+fCJNHC3C5qtaRw7ToXJvLniXwfg==",
-      "cpu": [
-        "x64"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "darwin"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-linux-arm-gnueabihf": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-arm-gnueabihf/-/core-linux-arm-gnueabihf-1.3.22.tgz",
-      "integrity": "sha512-4E+TdQT1oHnHjDaPs/DyrRy9lOuFd6ncEd67yYA4j9lFqt6nuz/jnXss45k8KU7wR5kOTtdW73xPwkU4NbOWdw==",
-      "cpu": [
-        "arm"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "linux"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-linux-arm64-gnu": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-arm64-gnu/-/core-linux-arm64-gnu-1.3.22.tgz",
-      "integrity": "sha512-6VcynOMbOBcbLutIPENI3Ejvg5LGz/Pwvzm25hM0FoiEtPxHA+tawQUwLx8Alk1Yr+Rnqid06UEZ0veJOGn2pQ==",
-      "cpu": [
-        "arm64"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "linux"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-linux-arm64-musl": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-arm64-musl/-/core-linux-arm64-musl-1.3.22.tgz",
-      "integrity": "sha512-86RxGy0L3qa4De3xWHx8vL2caTxvSLSWTlgUW/Yd4l1pvrCFibMjhkImGu5ViKiReX9DlBtJ7CBs4dln2kHidw==",
-      "cpu": [
-        "arm64"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "linux"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-linux-x64-gnu": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-x64-gnu/-/core-linux-x64-gnu-1.3.22.tgz",
-      "integrity": "sha512-FLkbiqsdXsVIFZi6iedx4rSBGX8x0vo/5aDlklSxJAAYOcQpO0QADKP5Yr65iMT1d6ABCt2d+/StpGLF7GWOcA==",
-      "cpu": [
-        "x64"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "linux"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-linux-x64-musl": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-x64-musl/-/core-linux-x64-musl-1.3.22.tgz",
-      "integrity": "sha512-giBuw+Z0Bq6fpZ0Y5TcfpcQwf9p/cE1fOQyO/K1XSTn/haQOqFi7421Jq/dFThSARZiXw1u9Om9VFbwxr8VI+A==",
-      "cpu": [
-        "x64"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "linux"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-win32-arm64-msvc": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-win32-arm64-msvc/-/core-win32-arm64-msvc-1.3.22.tgz",
-      "integrity": "sha512-loKGO+ZM2By6VdrmVJk1G79jVgDPaee93qLFuis5KyeoLLb4m1MlNMc/6SIDZUSuYg6NqaGP1spFeiFetMQ4Zg==",
-      "cpu": [
-        "arm64"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "win32"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-win32-ia32-msvc": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-win32-ia32-msvc/-/core-win32-ia32-msvc-1.3.22.tgz",
-      "integrity": "sha512-lvNWAZ3QjXMsrsch6oLLQVikT/hC/4ZcLrTBXa14HwQylaYigkGElgp3ekJr78HjWDPwB46GXwBbNMG0VNAfvA==",
-      "cpu": [
-        "ia32"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "win32"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
-    "node_modules/@swc/core-win32-x64-msvc": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-win32-x64-msvc/-/core-win32-x64-msvc-1.3.22.tgz",
-      "integrity": "sha512-ESyn4lZXAKEE3mcTaDfXatsolCiEfVGstsXdgBmZYa6o1IE1bDW8FE7Ob/Y+82WTpm9+A9ZYXYjZ62t67POHZg==",
-      "cpu": [
-        "x64"
-      ],
-      "dev": true,
-      "optional": true,
-      "os": [
-        "win32"
-      ],
-      "engines": {
-        "node": ">=10"
-      }
-    },
     "node_modules/@tootallnate/once": {
       "version": "2.0.0",
       "resolved": "https://registry.npmjs.org/@tootallnate/once/-/once-2.0.0.tgz",
@@ -6764,18 +6574,6 @@
         }
       }
     },
-    "node_modules/vite-plugin-top-level-await": {
-      "version": "1.2.2",
-      "resolved": "https://registry.npmjs.org/vite-plugin-top-level-await/-/vite-plugin-top-level-await-1.2.2.tgz",
-      "integrity": "sha512-JBuY9X5KCxZyzwQUOIBpgX0Jbe2BECxwElWPeyrdRC5xaWzDibTG65QBW25COmwqHBeVlV04u8GTlJkuOWhfgw==",
-      "dev": true,
-      "dependencies": {
-        "@swc/core": "^1.3.10"
-      },
-      "peerDependencies": {
-        "vite": ">=2.8"
-      }
-    },
     "node_modules/vitefu": {
       "version": "0.2.3",
       "resolved": "https://registry.npmjs.org/vitefu/-/vitefu-0.2.3.tgz",
@@ -7752,94 +7550,6 @@
         "vitefu": "^0.2.2"
       }
     },
-    "@swc/core": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core/-/core-1.3.22.tgz",
-      "integrity": "sha512-oQ9EPEb7NgWcGIDoVfLCuffvtC4MzVtrwjqwKzFHP8FUh1fn8+2wraOjkkDXW74BB4Hgve5ykkaHix9bebB9Ww==",
-      "dev": true,
-      "requires": {
-        "@swc/core-darwin-arm64": "1.3.22",
-        "@swc/core-darwin-x64": "1.3.22",
-        "@swc/core-linux-arm-gnueabihf": "1.3.22",
-        "@swc/core-linux-arm64-gnu": "1.3.22",
-        "@swc/core-linux-arm64-musl": "1.3.22",
-        "@swc/core-linux-x64-gnu": "1.3.22",
-        "@swc/core-linux-x64-musl": "1.3.22",
-        "@swc/core-win32-arm64-msvc": "1.3.22",
-        "@swc/core-win32-ia32-msvc": "1.3.22",
-        "@swc/core-win32-x64-msvc": "1.3.22"
-      }
-    },
-    "@swc/core-darwin-arm64": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-darwin-arm64/-/core-darwin-arm64-1.3.22.tgz",
-      "integrity": "sha512-MMhtPsuXp8gpUgr9bs+RZQ2IyFGiUNDG93usCDAFgAF+6VVp+YaAVjET/3/Bx5Lk2WAt0RxT62C9KTEw1YMo3w==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-darwin-x64": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-darwin-x64/-/core-darwin-x64-1.3.22.tgz",
-      "integrity": "sha512-SG6QbNat4GZ5VJU3Zo6a54oQOtbhJVE6BCQw4JjOCZJmAeBzNebGy9wsT4+fCJNHC3C5qtaRw7ToXJvLniXwfg==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-linux-arm-gnueabihf": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-arm-gnueabihf/-/core-linux-arm-gnueabihf-1.3.22.tgz",
-      "integrity": "sha512-4E+TdQT1oHnHjDaPs/DyrRy9lOuFd6ncEd67yYA4j9lFqt6nuz/jnXss45k8KU7wR5kOTtdW73xPwkU4NbOWdw==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-linux-arm64-gnu": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-arm64-gnu/-/core-linux-arm64-gnu-1.3.22.tgz",
-      "integrity": "sha512-6VcynOMbOBcbLutIPENI3Ejvg5LGz/Pwvzm25hM0FoiEtPxHA+tawQUwLx8Alk1Yr+Rnqid06UEZ0veJOGn2pQ==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-linux-arm64-musl": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-arm64-musl/-/core-linux-arm64-musl-1.3.22.tgz",
-      "integrity": "sha512-86RxGy0L3qa4De3xWHx8vL2caTxvSLSWTlgUW/Yd4l1pvrCFibMjhkImGu5ViKiReX9DlBtJ7CBs4dln2kHidw==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-linux-x64-gnu": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-x64-gnu/-/core-linux-x64-gnu-1.3.22.tgz",
-      "integrity": "sha512-FLkbiqsdXsVIFZi6iedx4rSBGX8x0vo/5aDlklSxJAAYOcQpO0QADKP5Yr65iMT1d6ABCt2d+/StpGLF7GWOcA==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-linux-x64-musl": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-linux-x64-musl/-/core-linux-x64-musl-1.3.22.tgz",
-      "integrity": "sha512-giBuw+Z0Bq6fpZ0Y5TcfpcQwf9p/cE1fOQyO/K1XSTn/haQOqFi7421Jq/dFThSARZiXw1u9Om9VFbwxr8VI+A==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-win32-arm64-msvc": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-win32-arm64-msvc/-/core-win32-arm64-msvc-1.3.22.tgz",
-      "integrity": "sha512-loKGO+ZM2By6VdrmVJk1G79jVgDPaee93qLFuis5KyeoLLb4m1MlNMc/6SIDZUSuYg6NqaGP1spFeiFetMQ4Zg==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-win32-ia32-msvc": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-win32-ia32-msvc/-/core-win32-ia32-msvc-1.3.22.tgz",
-      "integrity": "sha512-lvNWAZ3QjXMsrsch6oLLQVikT/hC/4ZcLrTBXa14HwQylaYigkGElgp3ekJr78HjWDPwB46GXwBbNMG0VNAfvA==",
-      "dev": true,
-      "optional": true
-    },
-    "@swc/core-win32-x64-msvc": {
-      "version": "1.3.22",
-      "resolved": "https://registry.npmjs.org/@swc/core-win32-x64-msvc/-/core-win32-x64-msvc-1.3.22.tgz",
-      "integrity": "sha512-ESyn4lZXAKEE3mcTaDfXatsolCiEfVGstsXdgBmZYa6o1IE1bDW8FE7Ob/Y+82WTpm9+A9ZYXYjZ62t67POHZg==",
-      "dev": true,
-      "optional": true
-    },
     "@tootallnate/once": {
       "version": "2.0.0",
       "resolved": "https://registry.npmjs.org/@tootallnate/once/-/once-2.0.0.tgz",
@@ -11767,15 +11477,6 @@
         "rollup": "^3.7.0"
       }
     },
-    "vite-plugin-top-level-await": {
-      "version": "1.2.2",
-      "resolved": "https://registry.npmjs.org/vite-plugin-top-level-await/-/vite-plugin-top-level-await-1.2.2.tgz",
-      "integrity": "sha512-JBuY9X5KCxZyzwQUOIBpgX0Jbe2BECxwElWPeyrdRC5xaWzDibTG65QBW25COmwqHBeVlV04u8GTlJkuOWhfgw==",
-      "dev": true,
-      "requires": {
-        "@swc/core": "^1.3.10"
-      }
-    },
     "vitefu": {
       "version": "0.2.3",
       "resolved": "https://registry.npmjs.org/vitefu/-/vitefu-0.2.3.tgz",
diff --git a/package.json b/package.json
index 0274d8b..4bd8faf 100644
--- a/package.json
+++ b/package.json
@@ -39,7 +39,6 @@
     "tslib": "^2.4.1",
     "typescript": "^4.9.4",
     "vite": "^4.0.0",
-    "vite-plugin-top-level-await": "^1.2.2",
     "vitest": "^0.25.7"
   },
   "dependencies": {
diff --git a/src/Markdown.svelte b/src/Markdown.svelte
index 894d16e..8681e45 100644
--- a/src/Markdown.svelte
+++ b/src/Markdown.svelte
@@ -1,9 +1,10 @@
 <script lang="ts">
   import type * as proj from "@app/project";
+  import type { MaybeHighlighted } from "@app/syntax";
 
   import dompurify from "dompurify";
   import matter from "@radicle/gray-matter";
-  import starry, { highlight } from "@app/syntax";
+  import { highlight } from "@app/syntax";
   import { base } from "@app/router";
   import {
     markdownExtensions as extensions,
@@ -71,11 +72,13 @@
         name.startsWith(prefix),
       );
       if (!className || !node.textContent) continue;
-      const tree = highlight(
-        starry,
-        node.textContent,
-        className.slice(prefix.length),
-      );
+      let tree: MaybeHighlighted;
+      (async () => {
+        tree = await highlight(
+          node.textContent ?? "",
+          className.slice(prefix.length),
+        );
+      })();
       if (tree) node.replaceChildren(toDom(tree, { fragment: true }));
     }
   });
diff --git a/src/base/projects/Blob.svelte b/src/base/projects/Blob.svelte
index 023aa84..0a59259 100644
--- a/src/base/projects/Blob.svelte
+++ b/src/base/projects/Blob.svelte
@@ -1,12 +1,13 @@
 <script lang="ts">
   import type { Blob } from "@app/project";
   import type { ProjectRoute } from "@app/router/definitions";
+  import type { MaybeHighlighted } from "@app/syntax";
 
   import dompurify from "dompurify";
   import HeaderToggleLabel from "@app/base/projects/HeaderToggleLabel.svelte";
   import ProjectLink from "@app/router/ProjectLink.svelte";
   import Readme from "@app/base/projects/Readme.svelte";
-  import starry, { highlight } from "@app/syntax";
+  import { highlight } from "@app/syntax";
   import { isMarkdownPath, scrollIntoView, twemoji } from "@app/utils";
   import { onMount } from "svelte";
   import { toHtml } from "hast-util-to-html";
@@ -49,7 +50,10 @@
     scrollIntoView(line);
   }
 
-  const content = highlight(starry, blob.content, fileExtension);
+  let content: MaybeHighlighted;
+  (async () => {
+    content = await highlight(blob.content, fileExtension);
+  })();
 </script>
 
 <style>
diff --git a/src/syntax/index.ts b/src/syntax/index.ts
index d0525e0..8230026 100644
--- a/src/syntax/index.ts
+++ b/src/syntax/index.ts
@@ -1,3 +1,4 @@
+import type { Root } from "@wooorm/starry-night";
 import { createStarryNight, common } from "@wooorm/starry-night";
 import sourceSvelte from "@wooorm/starry-night/lang/source.svelte.js";
 import sourceSolidity from "@wooorm/starry-night/lang/source.solidity.js";
@@ -6,6 +7,7 @@ import sourceErlang from "@wooorm/starry-night/lang/source.erlang.js";
 import sourceDockerfile from "@wooorm/starry-night/lang/source.dockerfile";
 import sourceAsciiDoc from "@wooorm/starry-night/lang/text.html.asciidoc";
 
+export type MaybeHighlighted = Root | undefined;
 type StarryNight = Awaited<ReturnType<typeof createStarryNight>>;
 
 // This takes 33 as common defined languages defined in node_modules/@wooorm/starry-night/lib/common.js
@@ -20,15 +22,18 @@ export const grammars = [
   sourceToml,
 ];
 
-export const highlight = (
-  starry: StarryNight,
+let st: StarryNight;
+
+export async function highlight(
   content: string,
   grammar: string,
-) => {
-  const scope = starry.flagToScope(grammar);
-  if (scope) {
-    return starry.highlight(content, scope);
+): Promise<MaybeHighlighted> {
+  if (st === undefined) {
+    st = await createStarryNight(grammars);
   }
-};
 
-export default await createStarryNight(grammars);
+  const scope = st.flagToScope(grammar);
+  if (scope) {
+    return st.highlight(content, scope);
+  }
+}
diff --git a/vite.config.ts b/vite.config.ts
index d2bc7e3..c1509cc 100644
--- a/vite.config.ts
+++ b/vite.config.ts
@@ -5,7 +5,6 @@ import type http from "node:http";
 
 import fs from "node:fs";
 import path from "node:path";
-import pluginTopLevelAwait from "vite-plugin-top-level-await";
 import { defineConfig } from "vite";
 import { svelte } from "@sveltejs/vite-plugin-svelte";
 
@@ -46,9 +45,6 @@ export default defineConfig({
     }),
     configureDevServer(),
     configurePreviewServer(),
-    // ES2020 supports top-level await, but Vite has an issue with it:
-    // https://github.com/vitejs/vite/issues/6985
-    pluginTopLevelAwait(),
   ],
   server: {
     // We have to set host here, otherwise CI binds to the ipv6 address and

```

Reply:
I honestly don't like these IIAFE that much, but I like the idea to be able to decouple the used grammars from `syntax/index.ts` and be able to include it according to each use case.

```diff
+  (async () => {
+    content = await highlight(blob.content, fileExtension);
+  })();
```